src/components/calcite-date-picker/utils.ts

```diff
@@ -36,10 +36,16 @@ export interface DateLocaleData {
 function getSupportedLocale(lang = "") {
   if (locales.indexOf(lang) > -1) {
     return lang;
-  } else {
-    const base = lang.split("-")[0];
-    return locales.indexOf(base) > -1 ? base : "en";
   }
+
+  lang = lang.toLowerCase();
```

Comment:
Just an idea, but we could simplify this even more if we use `String.replace`'s replacer function:

```ts
lang = lang.replace(/(\w+)-(\w+)/, (match, language, region) => `${language}-${region.toUpperCase()}`);

return locales.includes(lang) ? lang : "en";
```

I'll leave this up to you. 😉